server/src/services/restaurant.ts

```diff
@@ -177,6 +177,20 @@ async function findByYelpId(yelpId: string): Promise<{
   };
 }
 
+async function findFoodCategories(): Promise<string[]> {
+  const allFoodCategories: string[][] = await RestaurantModel.find({})
+    .select({
+      food_categories: 1,
+      _id: 0,
+    })
+    .then((result: any[]) => result.map((row: any) => row.food_categories));
+
+  // Construct set to remove duplicates
+  const foodCategories = [...new Set(allFoodCategories.flat())];
```

Comment:
flatten array from `string[][]` to `string[]` `array.flat()`
construct a set to remove duplicates from said array
then construct array back from set using spread operator

Note: using set takes linear time and memory

Alternative way is to sort the array so duplicates are next to each other and remove the duplicates iteratively.
This would take O(nlogn) time and linear memory - Using a set seems simpler unless I'm missing something